tools/crm_mon.c

```diff
@@ -804,21 +941,19 @@ main(int argc, char **argv)
 
     /* Extra sanity checks when in CGI mode */
     if (output_format == mon_output_cgi) {
-        argerr += (optind < argc);
-        argerr += (output_filename != NULL);
-        argerr += ((cib) && (cib->variant == cib_file));
-        argerr += (external_agent != NULL);
-        argerr += (daemonize == TRUE);  /* paranoia */
-
-    } else if (optind < argc) {
-        printf("non-option ARGV-elements: ");
-        while (optind < argc)
-            printf("%s ", argv[optind++]);
-        printf("\n");
-    }
-
-    if (argerr) {
-        return clean_up(CRM_EX_USAGE);
+        if (output_filename != NULL) {
+            fprintf(stderr, "CGI mode does not have an output filename set\n");
+            return clean_up(CRM_EX_USAGE);
+        } else if (cib && cib->variant == cib_file) {
+            fprintf(stderr, "CGI mode used with CIB file\n");
+            return clean_up(CRM_EX_USAGE);
+        } else if (options.external_agent != NULL) {
+            fprintf(stderr, "CGI mode requires --external-agent\n");
```

Comment:
cannot be used with

Reply:
Sorry, I don't quite follow what you mean here and in the next comment.  This was pretty much just a straight translation of the existing extra sanity checks block.

Reply:
if options.external_agent *!=* NULL

i.e. if it's supplied, it's an error -- the opposite of requires

Reply:
The original test (!=) was right, it's the message that's wrong :)

i.e. CGI mode *cannot* be used with an external agent